costmap_2d/plugins/obstruction_layer.cpp

```diff
@@ -240,46 +243,39 @@ void ObstructionLayer::onInitialize()
 
   obstruction_publisher_ = nh.advertise<costmap_2d::ObstructionListMsg>("obstructions", 1);
 
-  dsrv_ = NULL;
-  setupDynamicReconfigure(nh);
-}
-
-void ObstructionLayer::setupDynamicReconfigure(ros::NodeHandle& nh)
-{
-  dsrv_ = new dynamic_reconfigure::Server<costmap_2d::ObstructionPluginConfig>(nh);
-  dynamic_reconfigure::Server<costmap_2d::ObstructionPluginConfig>::CallbackType cb = boost::bind(
-      &ObstructionLayer::reconfigureCB, this, _1, _2);
-  dsrv_->setCallback(cb);
+  ROS_INFO("Starting config server for layer %s", name_.c_str());
```

Comment:
One issue I see is that we are no longer reading in params and configuring all of these values on startup.  The call to set the callback on the dynamic reconfigure server will pull the latest values from the param server.

Reply:
I added a function to initialize the values.